### ISSUE TITLE
enabling v2 telemetry with wasm results in `Unable to create Wasm HTTP filter`

### ISSUE THREAD

**Issue by ajohnstone** [istio/istio#42614]
### Bug Description

enabling v2 telemetry results in `Unable to create Wasm HTTP filter`

```
k -n istio-system logs -f deploy/istiod
2022-12-29T20:09:21.059758Z	warn	ads	ADS:LDS: ACK ERROR reviews-v3-c9c4fb987-whpm9.bookinfo-4 Internal:Error adding/updating listener(s) 10.96.70.143_443: Unable to create Wasm network filter
10.96.0.1_443: Unable to create Wasm network filter
10.96.188.98_443: Unable to create Wasm network filter
10.96.148.152_9402: Unable to create Wasm network filter
10.96.44.96_15012: Unable to create Wasm network filter
10.96.0.10_53: Unable to create Wasm network filter
10.96.44.96_443: Unable to create Wasm network filter
0.0.0.0_15014: Unable to create Wasm HTTP filter
10.96.70.143_15021: Unable to create Wasm HTTP filter
0.0.0.0_80: Unable to create Wasm HTTP filter
10.96.0.10_9153: Unable to create Wasm HTTP filter
0.0.0.0_15010: Unable to create Wasm HTTP filter
0.0.0.0_9080: Unable to create Wasm HTTP filter
virtualOutbound: Unable to create Wasm network filter
virtualInbound: Unable to create Wasm network filter

2022-12-29T20:09:22.024450Z	info	klog	successfully acquired lease istio-system/istio-leader
2022-12-29T20:09:22.024851Z	info	leader election lock obtained: istio-leader
2022-12-29T20:09:22.025233Z	info	Starting ingress controller
```

```
apiVersion: install.istio.io/v1alpha1
kind: IstioOperator
metadata:
  namespace: istio-system
  name: istiocontrolplane
spec:
  values:
...
    telemetry:
      enabled: true
      v2:
        enabled: true
        prometheus:
          wasmEnabled: true
        metadataExchange:
          wasmEnabled: true
...
```

### Version

```prose
istioctl version
client version: 1.16.1
control plane version: 1.16.1
data plane version: 1.16.1 (6 proxies)
```

```
kubectl version --short
Flag --short has been deprecated, and will be removed in the future. The --short output will become the default.
Client Version: v1.25.2
Kustomize Version: v4.5.7
Server Version: v1.25.3
```
```


### Additional Information

_No response_

**Comment by zirain** [comment]
cc @kyessenov

**Comment by zirain** [comment]
I cannot reproduce, can you share your steps?
did you use a customize proxyv2 image?

**Comment by howardjohn** [comment]
Is this even something we support now with the native stats rewrite?

**Comment by kyessenov** [comment]
Wasm stats filter will never go away as an example Wasm module, but we don't have to support operator configuration anymore.

**Comment by howardjohn** [comment]
This is fixed on newer versions, please upgrade and re-open if you still see it. Thanks

**Comment by GwonsooLee** [comment]
> This is fixed on newer versions, please upgrade and re-open if you still see it. Thanks

Hello @howardjohn. We are facing the same issue in istio version 1.7.1.

Here's the istio-proxy debug log.
```
2023-09-20T10:05:40.114281Z	warning	envoy config external/envoy/source/extensions/config_subscription/grpc/grpc_subscription_impl.cc:138	gRPC config for type.googleapis.com/envoy.config.listener.v3.Listener rejected: Error adding/updating listener(s) 172.20.43.67_443: Unable to create Wasm HTTP filter
172.20.153.48_3026: Unable to create Wasm HTTP filter
172.20.25.176_3023: Unable to create Wasm HTTP filter
172.20.153.48_3025: Unable to create Wasm HTTP filter
172.20.181.66_9001: Unable to create Wasm HTTP filter
172.20.233.59_8080: Unable to create Wasm HTTP filter
0.0.0.0_15010: Unable to create Wasm HTTP filter
172.20.50.244_3100: Unable to create Wasm HTTP filter
0.0.0.0_15014: Unable to create Wasm HTTP filter
172.20.25.176_3024: Unable to create Wasm HTTP filter
0.0.0.0_8080: Unable to create Wasm HTTP filter
172.20.187.17_443: Unable to create Wasm HTTP filter
0.0.0.0_9090: Unable to create Wasm HTTP filter
0.0.0.0_7979: Unable to create Wasm HTTP filter
0.0.0.0_5556: Unable to create Wasm HTTP filter
172.20.130.80_5558: Unable to create Wasm HTTP filter
172.20.108.136_8084: Unable to create Wasm HTTP filter
172.20.25.176_3026: Unable to create Wasm HTTP filter
172.20.25.176_5432: Unable to create Wasm HTTP filter
172.20.233.59_7000: Unable to create Wasm HTTP filter
172.20.108.136_8081: Unable to create Wasm HTTP filter
0.0.0.0_5557: Unable to create Wasm HTTP filter
172.20.183.255_8083: Unable to create Wasm HTTP filter
0.0.0.0_80: Unable to create Wasm HTTP filter
172.20.116.10_8082: Unable to create Wasm HTTP filter
172.20.177.5_15021: Unable to create Wasm HTTP filter
virtualInbound: Unable to create Wasm HTTP filter
```

But this issue is resolved if I upgraded the version to 1.19.0, as you suggested.

One thing I would like to know is the reason of the failure in the previous version...
Can you explain the reason why it was the problem in 1.17.1 version for us to understand the mechanism of istiod and envoy?

Thanks in advance!
